server/src/bge/System.cpp

```diff
@@ -199,9 +192,10 @@ namespace bge {
             forwardDirection.y = 0;
             forwardDirection.z = sin(glm::radians(req.yaw));
             forwardDirection = glm::normalize(forwardDirection);
-            req.forwardDirection = forwardDirection;
+            req.forwardDirection = glm::normalize(forwardDirection);
 
-            glm::vec3 rightwardDirection = glm::normalize(glm::cross(forwardDirection, glm::vec3(0, 1, 0)));
+            glm::vec3 rightwardDirection = glm::cross(forwardDirection, glm::vec3(0, 1, 0));
```

Comment:
Good catch, I didn't notice that forwardDirection and (0,1,0) are orthogonal so we don't need to do the extra normalization.